sherpa/astro/plot.py

```diff
@@ -42,6 +46,42 @@
            'FluxHistogram', 'EnergyFluxHistogram', 'PhotonFluxHistogram')
 
 
+# Identify "close-enough" bin edges when plotting histograms
+_tol = np.finfo(np.float32).eps
+
+
+def _check_hist_bins(plot):
+    """Ensure lo/hi edges that are "close" are merged.
+
+    Ensure that "close-enough" bin edges use the same value.  We do
+    this for all bins, even those that are identical, as it's
+    easier. The tolerance is taken to be the float32 "eps" setting, as
+    this seems to work for the (limited) data sets I've seen. This is
+    to fix issue #977.
+
+    Parameters
+    ----------
+    plot
+        The plot structure, which must have xlo and xhi attributes.
+
+    Notes
+    -----
+    Note that this holds even when plotting wavelength values, who
+    have xlo/xhi in decreasing order, since the lo/hi values still
+    hold.
+
+    """
+
+    # Technically idx should be 0 or 1, with no -1 values. We
+    # do not enforce this. What we do is to take all bins that
+    # appear similar (sao_fcmp==0) and set the xlo[i+1] bin
+    # to the xhi[i] value.
+    #
+    equal = sao_fcmp(plot.xlo[1:], plot.xhi[:-1], _tol)
```

Comment:
I'm fine with this as is, but I'm wondering why not do it with plain numpy?
```
equal = np.close(plot.xlo[1:], plot.xhi[:-1])
```
(could set `tol=_tol`, but default is already chosen to catch round-off errors for 32 bit systems).

Reply:
I did not know about np.close and I know about sao_fcmp...

Reply:
```sao_fcmp``` is more robust, checkout @DougBurke's comment:
[...which sao_fcmp thinks is the same, but numpy.digitize doesn't...](https://github.com/sherpa/sherpa/pull/756#issue-386719738)

Reply:
@dtnguyen2 The linked comment is about np.digitize which is made to digitize by looking for exactly identical values. `np.close` and its cousin `np.allclose` will tell you if two values are approximately the same with any tolerance you like (but default parameter are set to test for differences typical for float32 round-off errors). Yes, you could argue that there should be a `np.close_digitze` that performs the digitize operation using np.close, but that's not what's needed here.

Reply:
From the numpy doc:
[absolute(a - b) <= (atol + rtol * absolute(b))](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html)

subtracting two numbers that are close, I guess the numpy folks have a lot of numerical expert.  Whereas, the following comment is from gsl ```fcmp``` which ```sao_fcmp``` is built on to fix a few bugs.

>     /* x1 || x2 is 0 so the following substraction is kosher */

In numpy we trust?

Reply:
As the code works I'd rather not change anything at this time.

Reply:
I'm sure either is fine
If I want to be snarky then I could point out that 
> is build on to fix a few bugs

does not exactly inspire confidence in the gsl code...